### PR TITLE
Add the logout callback to prevent exception.

### PR DESCRIPTION
**Pull request by hcyuser**

### CODE REVIEW

**Review by hcyuser — COMMENTED** [review]
add callback for logout

**Comment by jackycute** [comment]
Hi @hcyuser,

Thanks for contribution.
Can you explain what is this PR solving?

**Comment by hcyuser** [comment]
If we don't add a callback in req.logout(callback),
It will get _Error: req#logout requires a callback function_ on the webpage when running node app.JS and clicking Sign out on the webpage.

**Comment by jackycute** [comment]
Thanks for adding this.
This is caused by https://stackoverflow.com/questions/72336177/error-reqlogout-requires-a-callback-function from https://github.com/jaredhanson/passport/issues/902#issuecomment-1133897569

File: lib/auth/index.js [hunk 1]
```diff
@@ -53,6 +53,8 @@ authRouter.get('/logout', function (req, res) {
   if (config.debug && req.isAuthenticated()) {
     logger.debug('user logout: ' + req.user.id)
   }
-  req.logout()
+  req.logout(()=>{
+    logger.debug('Please implement logout callback here.')
+  })
```
Comment (jackycute) [inline lib/auth/index.js:58]: It should be like:

```suggestion
  req.logout((err) => {
    if (err) { return next(err) }
    res.redirect(config.serverURL + '/')
  })
```

and add `next` as the last parameter for get route callback function

**Review by brent85 — APPROVED** [review]

File: lib/auth/index.js (hunk 1 above)
Comment (jackycute) [reply lib/auth/index.js:58]: Hi @hcyuser,
Please consider this suggestion.

**Comment by Yukaii** [comment]
closed in favor of #1837

**Pull request closed without merging**